src/models/GraphQLModels.ts

```diff
@@ -0,0 +1,101 @@
+export type GraphQLInput = {
+  // pagination
+  limit?: number;
+  cursor?: string;
+
+  // sorting
+  sortKey?: string | string[];
+};
+
+export type GranulesInput = GraphQLInput & {
+  // filtering
+  provider?: string;
+
+  // ids
+  conceptIds?: string[];
+  readableGranuleName?: string[];
+
+  // collections
+  collectionConceptIds?: string[];
+
+  // bbox
+  boundingBox?: string;
+
+  // intersects
+  polygon?: string[];
+  line?: string[];
+  point?: string[];
+
+  // datetime
+  temporal?: string;
+
+  // extensions
+  cloudCover?: {
+    min?: number;
+    max?: number;
+  };
+};
+
+export type CollectionsInput = GraphQLInput & {
+  // filtering
+  provider?: string;
+  providers?: string[];
+  conceptIds?: string[];
+  cloudHosted?: boolean;
+  hasGranules?: boolean;
+  includeFacets?: string;
+};
+
+export type FacetFilter = {
+  title: string;
+  type: string;
+  applied: boolean;
+  count: number;
+  links: {
+    apply?: string;
+    remove?: string;
+  };
+  hasChildren: boolean;
+  children?: FacetFilter[];
+};
+
+export type FacetGroup = {
+  title: string;
+  type: string;
+  hasChildren: boolean;
+  children: FacetFilter[] | FacetGroup[];
+};
+
+export type Collection = {
+  shortName: string;
```

Comment:
Do these fields need to have an exception to be null since they aren’t required fields in graphql? So they could return null but, I don’t think that CMR will let you ingest collections without those fields so there shouldn't be any records like that